backend/src/emissions/tests/test_plantrip.py

```diff
@@ -21,42 +21,169 @@
+
+
+@pytest.mark.parametrize('transportation_mode', ['Train'])
+@pytest.mark.parametrize('fuel_type', ['diesel', 'electric', 'average'])
+@pytest.mark.parametrize('roundtrip', [True, False])
+def test_plan_trip_train(transportation_mode, fuel_type, roundtrip):
+    """Test whether trip planner throws error for different parameter combinations"""
+    query = """
+        mutation planTrip ($transportationMode: String!, $fuelType: String!, $roundtrip: Boolean!) {
+            planTrip (input: {
+                transportationMode: $transportationMode
+                fuelType: $fuelType
+                roundtrip: $roundtrip
+                distance: 200
+            }) {
+                success
+                message
+                co2e
+            }
+        }
+    """
+
+    headers = {
+        "Content-Type": "application/json",
+    }
+    variables = {"transportationMode": transportation_mode, "fuelType": fuel_type, "roundtrip": roundtrip}
+    response = requests.post(GRAPHQL_URL, json={"query": query, "variables": variables}, headers=headers)
+    logger.warning(response.content)
+    assert response.status_code == 200
+    data = response.json()
+    logger.warning(data)
+    assert data["data"]["planTrip"]["success"]
+
 
+@pytest.mark.parametrize('transportation_mode', ['Plane'])
+@pytest.mark.parametrize('seating_class', ['average', 'Economy class', 'Premium economy class', 'Business class', 'First class'])
+@pytest.mark.parametrize('roundtrip', [True, False])
+def test_plan_trip_plane(transportation_mode, seating_class, roundtrip):
+    """Test whether trip planner throws error for different parameter combinations"""
+    query = """
+        mutation planTrip ($transportationMode: String!, $seatingClass: String!, $roundtrip: Boolean!) {
+            planTrip (input: {
+                transportationMode: $transportationMode
+                seatingClass: $seatingClass
+                roundtrip: $roundtrip
+                distance: 500
+            }) {
+                success
+                message
+                co2e
+            }
+        }
+    """
+
+    headers = {
+        "Content-Type": "application/json",
+    }
+    variables = {"transportationMode": transportation_mode, "seatingClass": seating_class, "roundtrip": roundtrip}
+    response = requests.post(GRAPHQL_URL, json={"query": query, "variables": variables}, headers=headers)
+    logger.warning(response.content)
+    assert response.status_code == 200
+    data = response.json()
+    logger.warning(data)
+    assert data["data"]["planTrip"]["success"]
+
+
+@pytest.mark.parametrize('transportation_mode', ['Ferry'])
+@pytest.mark.parametrize('seating_class', ['average', 'Foot passenger', 'Car passenger'])
+@pytest.mark.parametrize('roundtrip', [True, False])
+def test_plan_trip_ferry(transportation_mode, seating_class, roundtrip):
+    """Test whether trip planner throws error for different parameter combinations"""
```

Comment:
Do all of these tests have the same name? Or am I reading this wrong?

Reply:
Nvm, they have the same comment. I was reading the comment. Which I think should be changed for each specific test. (although if we make the test names more descriptive, the comment isn't necessary)

Reply:
I've adjusted the doc strings